internal/dag/conditions.go

```diff
@@ -22,11 +22,11 @@ import (
 	projcontour "github.com/projectcontour/contour/apis/projectcontour/v1"
 )
 
-// mergePathConditions merges the given slice of prefix Conditions into a single
+// mergePathMatchConditions merges the given slice of prefix MatchConditions into a single
 // prefix Condition.
-// pathConditionsValid guarantees that if a prefix is present, it will start with a
+// pathMatchConditionsValid guarantees that if a prefix is present, it will start with a
 // / character, so we can simply concatenate.
-func mergePathConditions(conds []projcontour.Condition) Condition {
+func mergePathMatchConditions(conds []projcontour.MatchCondition) MatchCondition {
```

Comment:
Should this have been called `mergePrefixConditions` from the start? So now it would be `mergePrefixMatchConditions`?

Just a little confusing if we do add true "path" matching later on.

Reply:
Probably, yeah.

Reply:
Actually, I looked into this, and I think it's better to leave it the same, and extend this function with additional path matching merge logic if we add non-prefix matching later.

Reply:
If we're going to do that, I'd rather it be in a separate PR, because we'll need to rename a bunch of other stuff from Path to Prefix as well.